File: Cocoa/Plug-Ins/Template Editor/ElementDVDR.m
#import "ElementDVDR.h"

@implementation ElementDVDR

- (instancetype)initForType:(NSString *)t withLabel:(NSString *)l
{
    if (self = [super initForType:t withLabel:l]) {
        self.rowHeight = l.length ? 17 : 1;
    }
    return self;
}

- (void)configureGroupView:(NSTableCellView *)view
{
    view.textField.stringValue = self.displayLabel;
}

@end

File: ResForge/Template Editor/Elements/ElementDVDR.swift
import Cocoa

class ElementDVDR: Element, GroupElement {
    required init(type: String, label: String) {
        super.init(type: type, label: label)
        if label.isEmpty {
            rowHeight = 1
        } else {
            rowHeight = Double(label.components(separatedBy: "\n").count * 15) + 1
        }
    }

    func configureGroup(view: NSTableCellView) {
        view.textField?.stringValue = label
    }
}
